src/common/components/repository-row/index.js

```diff
@@ -201,14 +201,23 @@ export class RepositoryRowView extends Component {
       });
     }
 
-    const nextSnapcraftData = nextProps.snap.snapcraftData;
-
-    if (snapNameIsMismatched(nextProps.snap)) {
-      this.checkNameOwnership(nextProps.nameOwnership, nextSnapcraftData.name);
-    }
-
-    if (nextProps.snap.storeName) {
-      this.checkNameOwnership(nextProps.nameOwnership, nextProps.snap.storeName);
+    // only attempt checking name ownership if user is authenticated
+    // and we are not fetching any auth data (because fetched account info will
+    // contain names already registered by given user)
+    if (this.props.authStore.authenticated &&
+        !nextProps.authStore.isFetching && !this.props.authStore.isFetching) {
```

Comment:
If authStore data is fetching (possibly waiting for `/store/account` response) we shouldn't attempt any name check, because account info didn't yet pre-populate the list.